tests/ui/async-await/return-type-notation/issue-110963-early.stderr

```diff
@@ -35,17 +30,11 @@ LL | |         if !hc.check().await {
 LL | |             log_health_check_failure().await;
 LL | |         }
 LL | |     });
-   | |______^ one type is more general than the other
-   |
-   = note: expected trait `Send`
-              found trait `for<'a> Send`
-note: the lifetime requirement is introduced here
-  --> $DIR/issue-110963-early.rs:34:17
+   | |______^ implementation of `Send` is not general enough
    |
-LL |     F: Future + Send + 'static,
-   |                 ^^^^
```

Comment:
Could we recover this easily?

Reply:
looked into this a bit, we could extend `NiceRegionError::report_trait_placeholder_mismatch` to also point to the obligation cause, I do not want to do so in this PR however.